Sources/xcproj/PBXTarget.swift

```diff
@@ -95,7 +95,12 @@ public class PBXTarget: PBXObject {
                 let comment = proj.objects.buildPhaseName(buildPhaseReference: buildPhase)
                 return .string(CommentedString(buildPhase, comment: comment))
         })
-        dictionary["buildRules"] = .array(buildRules.map {.string(CommentedString($0))})
+
+        // Xcode doesn't write empty PBXLegacyTarget buildRules
+        if !(self is PBXLegacyTarget) || !buildRules.isEmpty {
```

Comment:
Good catch 👍

Reply:
It's quite weird behaviour from Xcode...